node/src/accountant/scanners/mod.rs

```diff
@@ -377,65 +377,74 @@ impl PayableScanner {
-            .copied()
-            .collect::<Vec<&PendingPayable>>();
+            .map(|payable| (payable.hash, &payable.recipient_wallet))
+            .collect::<HashMap<H256, &Wallet>>();
 
-        let rowid_pairs_sorted_by_hashes = self
-            .pending_payable_dao
-            .fingerprints_rowids(&hashes)
-            .into_iter()
-            .sorted_by(|(_, hash_a), (_, hash_b)| Ord::cmp(&hash_a, &hash_b))
-            .collect::<Vec<(Option<u64>, H256)>>();
+        let transaction_hashes = self.pending_payable_dao.fingerprints_rowids(&hashes);
+        let mut rowid_pairs_with_hashes = transaction_hashes
+            .rowid_results
+            .iter()
+            .map(|(_rowid, hash)| *hash)
+            .collect::<HashSet<H256>>();
+        for hash in &transaction_hashes.no_rowid_results {
+            rowid_pairs_with_hashes.insert(*hash);
+        }
+        let sent_payables_hashes = hashes.iter().copied().collect::<HashSet<H256>>();
 
-        if !Self::is_symmetrical(
-            &sent_payables_sorted_by_hashes,
-            &rowid_pairs_sorted_by_hashes,
-        ) {
+        if !PayableScanner::is_symmetrical(sent_payables_hashes, rowid_pairs_with_hashes) {
             panic!(
                 "Inconsistency in two maps, they cannot be matched by hashes. Data set directly \
                 sent from BlockchainBridge: {:?}, set derived from the DB: {:?}",
-                sent_payables_sorted_by_hashes, rowid_pairs_sorted_by_hashes
+                sent_payables, transaction_hashes.rowid_results
             )
         }
 
-        rowid_pairs_sorted_by_hashes
+        let pending_payables_with_rowid = transaction_hashes
+            .rowid_results
+            .into_iter()
+            .map(|(rowid, hash)| {
+                PendingPayableMetadata::new(
+                    <&Wallet>::clone(
+                        sent_payables_hashmap
+                            .get(&hash)
+                            .expect("expect transaction hash, but it disappear"),
+                    ),
+                    hash,
+                    Some(rowid),
+                )
+            })
+            .collect_vec();
+        let pending_payables_without_rowid = transaction_hashes
+            .no_rowid_results
             .into_iter()
-            .zip(sent_payables_sorted_by_hashes.into_iter())
-            .map(sew_metadata)
-            .partition(|pp_triple| pp_triple.rowid_opt.is_some())
+            .map(|hash| {
+                PendingPayableMetadata::new(
+                    <&Wallet>::clone(
+                        sent_payables_hashmap
+                            .get(&hash)
+                            .expect("expect transaction hash, but it disappear"),
+                    ),
+                    hash,
+                    None,
+                )
+            })
+            .collect_vec();
+        (pending_payables_with_rowid, pending_payables_without_rowid)
```

Comment:
You can consider to have a struct with two named fields for clearing up the kind of the set for readers bumping into it outside this function. Not like two anonymous items in a tuple.

Reply:
I have been thinking about it a lot, but still, I see that tuple of `(existent, nonexistent)` is a pretty straightforward data structure, in case we have fn `separate_existent_and_nonexistent_fingerprints`